cluster-autoscaler/cloudprovider/gce/os_reserved.go

```diff
@@ -16,13 +16,23 @@ limitations under the License.
 
 package gce
 
+// MigOsInfo store os parameters.
+type MigOsInfo interface {
+	// Os return operating system.
+	Os() OperatingSystem
+	// OsDistribution return operating system distribution.
+	OsDistribution() OperatingSystemDistribution
+	// Arch return system architecture.
+	Arch() SystemArchitecture
+}
+
 // OsReservedCalculator calculates the OS reserved values.
 type OsReservedCalculator interface {
 	// CalculateKernelReserved computes how much memory OS kernel will reserve.
 	// NodeVersion parameter is optional. If empty string is passed a result calculated using default node version will be returned.
-	CalculateKernelReserved(physicalMemory int64, os OperatingSystem, osDistribution OperatingSystemDistribution, arch SystemArchitecture, nodeVersion string) int64
+	CalculateKernelReserved(m MigOsInfo, physicalMemory int64, nodeVersion string) int64
```

Comment:
I think now we can remove the nodeVersion as well, both from these functions and from Mig interface. It is not needed in the GCE implementation, and we are now able to add this in internal fork one using MigOsInfo.

Reply:
Right, done